org.scala-ide.sdt.core/src/org/scalaide/ui/editor/DefaultScalaEditorConfiguration.scala

```diff
@@ -0,0 +1,107 @@
+package org.scalaide.ui.editor
+
+import org.eclipse.jface.text.source.IAnnotationHover
+import org.eclipse.jface.preference.IPreferenceStore
+import org.eclipse.jface.text.reconciler.IReconciler
+import org.eclipse.jface.text.reconciler.MonoReconciler
+import org.eclipse.jface.text.source.DefaultAnnotationHover
+import org.scalaide.core.lexical.ScalaPartitions
+import org.eclipse.jface.text.presentation.PresentationReconciler
+import org.scalaide.core.lexical.ScalaCodeScanners
+import org.eclipse.jface.text.source.SourceViewerConfiguration
+import org.eclipse.jface.text.information.InformationPresenter
+import org.eclipse.jdt.ui.text.IJavaPartitions
+import org.eclipse.jface.text.IDocumentListener
+import org.eclipse.jface.text.source.ISourceViewer
+import org.eclipse.jface.text.IDocument
+import org.scalaide.ui.editor.hover.IScalaHover
+import org.eclipse.jface.text.ITextHover
+import org.scalaide.core.IScalaPlugin
+import org.eclipse.jface.text.DocumentEvent
+import org.eclipse.jface.util.PropertyChangeEvent
+import org.eclipse.jface.text.rules.DefaultDamagerRepairer
+import org.eclipse.ui.texteditor.ChainedPreferenceStore
+
+/** A default SourceViewerConfiguration class for Scala-based editors.
+ *
+ *  This class can be subclassed. It provides a default configuration including
+ *  reconciliation (errors-as-you-type), hovers and Scala syntax highlighting.
+ *
+ *  The workhorse of this implementation is [[InteractiveCompilationUnit]], which provides
+ *  the mapping between original source and Scala-source.
+ *
+ *  Any method can be overwritten.
+ *
+ *  @note Standard content assist is missing from this configuration, but should be easy to add
+ *        in a subsequent release
+ *
+ *  @since 4.3.0
+ */
+trait DefaultScalaEditorConfiguration extends SourceViewerConfiguration {
```

Comment:
BTW, I think the existing `ScalaSourceViewerConfiguration` should subclass this class. But I don't think we can assume all Scala-based editors can inherit *all* of the existing Scala configuration. The Scala editor inherits the Java one, and some things will be done differently between Scala and Sbt files (or Play templates). I don't see how they can all share the exact same class, but if you manage, that's be great. But it's way easier to just refactor common configuration into this one, and inherit it elsewhere.